include/net/fota_download.h

```diff
@@ -146,6 +146,28 @@ int fota_download(const char *host, const char *file, const int *sec_tag_list,
 		  uint8_t sec_tag_count, uint8_t pdn_id, size_t fragment_size,
 		  const enum dfu_target_image_type expected_type);
 
+/**@brief Download the given file with the specified image type from the given host.
+ *
+ * Validate that the file type matches the expected type before proceeding with the download.
+ * When the download is complete, the secondary slot of MCUboot is tagged as having
+ * valid firmware inside it. The completion is reported through an event.
+ *
+ * URI parameters (host and file) are not copied, so pointers must stay valid
+ * until download is finished.
+ *
+ * @param host Name of host to start downloading from. Can include scheme
+ *             and port number, for example https://google.com:443
+ * @param file Path to the file you wish to download. See fota_download_any()
+ *             for details on expected format.
+ * @param config Download client configuration options.
+ * @param expected_type Type of firmware file to be downloaded and installed.
+ *
+ * @retval 0	     If download has started successfully.
+ * @retval -EALREADY If download is already ongoing.
+ *                   Otherwise, a negative value is returned.
+ */
+int fota_download_v2(const char *host, const char *file, const struct download_client_cfg *config,
```

Comment:
```suggestion
int fota_download_cfg(const char *host, const char *file, const struct download_client_cfg *config,
```

We don't version our APIs other than what the SDK version is.
So I recommend to use just `_cfg` postfix instead of `_v2`

Reply:
I would suggest below:
.define a new Kconfig, e.g. `CONFIG_FOTA_DOWNLOAD_NEW_API`, which is default 'y'. We use this flag to clean up all legacy APIs and provide a new API of `fota_download()`, apply it to all known clients in NCS;
.mark all old APIs to be deprecated in next NCS version and remove them in next next NCS version;
.Remove the Kconfig in next next NCS version.